nservicebus/messaging/databus.md

```diff
@@ -68,7 +68,25 @@ snippet:MessageWithLargePayloadUsingConvention
 
 NServiceBus DataBus implementations currently behave differently with regard to cleanup of physical attachments used to transfer data properties. `FileShareDataBus` **does not** remove physical attachments once the message is gone. `AzureDataBus` **does** remove Azure storage blobs used for physical attachments once the message is gone.
 
+### Custom cleanup strategy
 
+NServiceBus does not remove the transferred message attachments once a message is delivered. Automatically removing these attachments can cause problems in many situations. For example:
+
+- The message can be (functionally) deferred so that the file will be processed later. Removing the file after deferring the message, results in a message without the corresponding file.
+- Functional requirements might dictate the message to be available for a longer duration.
+- The file system does not participate in distributed transactions. If for some reason, the message handler throws an exception and the transaction rolls back, the file delete operation on the attachment cannot be rolled back. Therefore, when the message is retried, the attachment will no longer be present causing additional problems.
+- If the outbox feature in NServiceBus is enabled, the message will be removed from the incoming queue, but it might not have been processed yet. 
+
+The business requirements should specify when the file should be processed and when it can be removed. Therefore the business should come up with a strategy to remove the files. An option is a strategy like removing the file after (Max(SLA) + x-days), so that it is unlikely that the file still hasn’t been processed.
```

Comment:
>The business requirements should specify when the file should be processed and when it can be removed. Therefore the business should come up with a strategy to remove the files. An option is a strategy like removing the file after (Max(SLA) + x-days), so that it is unlikely that the file still hasn’t been processed.

This paragraph is redundant as the following paragraph has the updated information.